src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -293,6 +296,9 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
                     ImportType(type, mappings, importedTypes, warnings, importer, parsableerrors);
                 }
             }
+            string nameOrLocation = assembly.Location;
+            if (nameOrLocation == string.Empty)
```

Comment:
Should this also check for `null`?

Reply:
The only implementation I found in the runtime was in [RuntimeAssembly.cs](https://github.com/dotnet/runtime/blob/main/src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs#L457) and suppresses the warning for nullable. Although I'm not very sure if the [Qcall ](https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/assemblynative.cpp#L410) will not return null

Reply:
I change it either way to string.IsNullOrEmpty pattern

Reply:
`Assembly.Location` says it returns non-null string. Unless there's any particular reason to think there's a bug, I think we can trust that this never returns null.